### PR TITLE
account for null terminator

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**

### CODE REVIEW

**Comment by kaleb-himes** [comment]
ok to test

File: wrapper/CSharp/wolfSSL_CSharp/wolfSSL.cs [hunk 1]
```diff
@@ -569,13 +574,35 @@ public static int read(IntPtr ssl, StringBuilder buf, int sz)
...
+                {
+                    /* Get data that was sent accross and store it using a literal read of
+                     * the conversion from bytes to character. Takes care of if
+                     * a null terminator is part of the message read.
+                     */
+                    msg = new byte[ret];
+                    Marshal.Copy(data, msg, 0, ret);
+                    for (int i = 0; i < ret; i++)
+                    {
+                        buf.Append(@Convert.ToChar(msg[i]));
```
Comment (dgarske) [inline wrapper/CSharp/wolfSSL_CSharp/wolfSSL.cs]: Hey Jacob, worried about the Convert.ToChar(byte) handling non-printable data correctly. Manual says "Converts the value of the specified 8-bit unsigned integer to its equivalent Unicode character". Is this "read" version of the API meant for just string data and if reading binary data would use the other byte[] version of the "read" API below?

**Comment by dgarske** [comment]
Otherwise looks great! Thanks for making that fix/enhancement.

**Comment by kaleb-himes** [comment]
ok to test

**Comment by JacobBarthelmeh** [comment]
Hi David, 
Is a valid worry. Though non printable they are still getting stored in the StringBuilder object and can be accessed by something like System.Text.Encoding.Unicode.GetBytes(buffer.ToString()). I've tested this on an array containing [0,255] being read in from the socket. If wanting to work with raw bytes this can cause a headache because char can be two bytes so the second is always 0. The overloaded function using a byte[] is for those who wish to avoid the necessity of performing conversions when working with raw bytes. Your comment and testing did lead me to make a change in "write" from UTF8 to Default encoding.

**Pull request merged**